File: packages/permissions/ios/Plugin/Classes/Results/RequestResult.swift
import Foundation
import Capacitor

@objc public class RequestResult: NSObject, Result {
    let statuses: [PermissionStatus]

    init(statuses: [PermissionStatus]) {
        self.statuses = statuses
    }

    @objc public func toJSObject() -> AnyObject {
        var statusesResult = JSArray()
        for status in statuses {
            if let statusObject = status.toJSObject() as? JSObject {
                statusesResult.append(statusObject)
            }
        }
        var result = JSObject()
        result["statuses"] = statusesResult
        return result as AnyObject
    }
}

File: packages/permissions/android/src/main/java/io/capawesome/capacitorjs/plugins/permissions/classes/results/RequestResult.java
package io.capawesome.capacitorjs.plugins.permissions.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.permissions.interfaces.Result;
import java.util.List;

public class RequestResult implements Result {

    @NonNull
    private final List<PermissionStatus> statuses;

    public RequestResult(@NonNull List<PermissionStatus> statuses) {
        this.statuses = statuses;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSArray statusesResult = new JSArray();
        for (PermissionStatus status : statuses) {
            statusesResult.put(status.toJSObject());
        }
        JSObject result = new JSObject();
        result.put("statuses", statusesResult);
        return result;
    }
}
